src/baldr/graphtile.cc

```diff
@@ -155,19 +141,18 @@ graph_tile_ptr GraphTile::Create(const GraphId& graphid,
 GraphTile::GraphTile(const GraphId& graphid,
                      std::unique_ptr<const GraphMemory> memory,
                      std::unique_ptr<const GraphMemory> traffic_memory)
-    : header_(nullptr), traffic_tile(std::move(traffic_memory)) {
+    : header_(nullptr), traffic_tile(std::move(traffic_memory)),
+      ref_counter_(new ref_counter{false}) {
```

Comment:
@dnesbitt61 if you dont want to wait for me to finish this pr before you test, you can also change this to `true` and all the  graph tile ptrs should be thread safe in terms of managing their lifetimes